pkg/cmd/stack/destroy.go

```diff
@@ -15,27 +15,110 @@ package stack
 
 import (
 	"context"
+	"encoding/base64"
 	"fmt"
 	"strings"
+	"time"
 
 	"github.com/okteto/okteto/cmd/utils"
-	"github.com/okteto/okteto/pkg/helm"
+	"github.com/okteto/okteto/pkg/k8s/client"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	"github.com/okteto/okteto/pkg/k8s/deployments"
+	okLabels "github.com/okteto/okteto/pkg/k8s/labels"
+	"github.com/okteto/okteto/pkg/k8s/pods"
+	"github.com/okteto/okteto/pkg/k8s/services"
+	"github.com/okteto/okteto/pkg/k8s/statefulsets"
+	"github.com/okteto/okteto/pkg/k8s/volumes"
 	"github.com/okteto/okteto/pkg/model"
 	"helm.sh/helm/v3/pkg/action"
 	"helm.sh/helm/v3/pkg/cli"
+	"k8s.io/client-go/kubernetes"
 )
 
 //Destroy destroys a stack
-func Destroy(ctx context.Context, s *model.Stack) error {
+func Destroy(ctx context.Context, s *model.Stack, removeVolumes bool) error {
+	if s.Namespace == "" {
+		s.Namespace = client.GetContextNamespace("")
+	}
+
+	c, _, _ := client.GetLocal()
+
+	cfg := translateConfigMap(s)
+	output := fmt.Sprintf("Destroying stack '%s'...", s.Name)
+	cfg.Data[statusField] = destroyingStatus
+	cfg.Data[outputField] = base64.StdEncoding.EncodeToString([]byte(output))
+	if err := configmaps.Deploy(ctx, cfg, s.Namespace, c); err != nil {
+		return err
+	}
+
+	err := destroy(ctx, s, removeVolumes, c)
+	if err != nil {
+		output = fmt.Sprintf("%s\nStack '%s' destruction failed: %s", output, s.Name, err.Error())
+		cfg.Data[statusField] = errorStatus
+		cfg.Data[outputField] = base64.StdEncoding.EncodeToString([]byte(output))
+		if err := configmaps.Deploy(ctx, cfg, s.Namespace, c); err != nil {
+			return err
+		}
+	} else {
+		if err := configmaps.Destroy(ctx, cfg.Name, s.Namespace, c); err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+func destroy(ctx context.Context, s *model.Stack, removeVolumes bool, c *kubernetes.Clientset) error {
 	spinner := utils.NewSpinner(fmt.Sprintf("Destroying stack '%s'...", s.Name))
 	spinner.Start()
 	defer spinner.Stop()
 
+	if err := destroyHelmRelease(ctx, s, spinner); err != nil {
+		return err
+	}
+
+	s.Services = nil
+	if err := destroyServicesNotInStack(ctx, s, c); err != nil {
+		return err
+	}
+
+	spinner.Update("Waiting for services to be destroyed...")
+	if err := waitForPodsToBeDestroyed(ctx, s, c); err != nil {
+		return err
+	}
+
+	if removeVolumes {
+		spinner.Update("Destroying volumes...")
+		if err := destroyStackVolumes(ctx, s, c); err != nil {
+			return err
+		}
+	}
+
+	if err := configmaps.Destroy(ctx, s.GetConfigMapName(), s.Namespace, c); err != nil {
```

Comment:
can we do the volumes at the end instead of the configmap?

Reply:
the configmap must be the last one, it is the stack metadata